### ISSUE TITLE
Training on custom data

### ISSUE THREAD

**Issue by nicholasmireles** [yikang-li/FactorizableNet#9]
Hello!
I've been looking into how to train this model on an external dataset, and I was hoping to get some clarification on the JSON files you feed the data loaders. I imagine the kmeans_anchors file is generated from the RPN model that you use, but could I get some insight into what exactly is in the inverse_weights file? I can tell that you're weighting the object and predicate classes by some value, but how exactly were those values generated? Or would I need them at all if I make a data loader class for my dataset?

**Comment by yikang-li** [comment]
kmeans_anchors: generating the anchors with k-means clustering on the object boxes (in log scale) of the training set (all the images are normalized to the input size, the longer sides are normalized to 600). 

inverse_weights (it's an empirical setting. I forgot the exact formula we use, it's like this one): sqrt(max_class_frequency / class_frequency), which gives more weights to the classes with fewer instances.

They are just from our previous experience. We didn't do any ablation study on the effects of these settings. 

Hope it helps. 

Best,
Yikang

**Comment by PRYUS** [comment]
Hey @nicholasmireles !
I and my colleague have also been looking into how to train this model on an external dataset, and we were hoping to get some help on it. Could you please help us?

**Comment by fomalhaut-b** [comment]
@yikang-li I second this. Would it be possible to add a short update in README on how to (i) train on custom data and (ii) use pretrained model on custom data to get output. 

Much thanks!
